src/index.js

```diff
@@ -22,12 +23,34 @@ const options = {
  */
 module.exports = function findLocalDevices (address) {
   var key = String(address)
-  lock[key] =
-    lock[key] ||
-    (address
-      ? pingServer(address).then(arpOne)
-      : pingServers().then(arpAll)
-    ).then(unlock(key))
+
+  if (address && new RegExp('/|-').test(address)) {
+    try {
+      servers = getIPRange(key)
+    } catch (error) {
+      // Note: currently doesn't throw the intended error message from the package maintainer
+      // It will still be caught, PR here though: https://github.com/JoeScho/get-ip-range/pull/6
+      return error
+    }
+  } else {
+    servers = getServers()
+  }
+
+  if (!lock[key]) {
+    if (address) {
+      if (new RegExp('/|-').test(key)) {
+        // range of IPs detected
+        lock[key] = arpAll().then(unlock(key))
```

Comment:
```suggestion
        lock[key] = pingServers().then(arpAll).then(unlock(key))
```

Don't we need to invoke `pingServers` (Sends a ping to all servers to update the arp table.) first, as well here?

Then you could simplify this if-block, as the `// no address given` does the same thing.

Reply:
Indeed: https://github.com/DylanPiercey/local-devices/blob/a27a69d1779ea7d2d77cddabeaeb722e70ccd8fc/src/index.js#L29